crates/storage/db/src/tables/models/blocks.rs

```diff
@@ -12,29 +13,62 @@ pub type NumTransactions = u64;
 /// transaction in the block and the total number of transactions
 #[derive(Debug, Default, Eq, PartialEq, Clone)]
 #[main_codec]
-pub struct StoredBlockBody {
-    /// The id of the first transaction in this block
-    pub start_tx_id: TxNumber,
+pub struct StoredBlockMeta {
+    /// The number of the first transaction in this block
+    ///
+    /// Note: If the block is empty, this is the number of the first transaction
+    /// in the next non-empty block.
+    pub first_tx_num: TxNumber,
+    /// The id of the first transition in this block.
+    ///
+    /// Note: If the block is empty, this is the id of the first transition
+    /// in the next non-empty block.
+    pub first_transition_id: TransitionId,
     /// The total number of transactions in the block
+    ///
+    /// Note: Number of transitions is equal to number of transactions with
+    /// additional transition for block change if block has block reward or withdrawal.
     pub tx_count: NumTransactions,
+    /// Flags if there is additional transition changeset of the withdrawal or block reward.
+    pub has_block_change: bool,
 }
 
-impl StoredBlockBody {
-    /// Return the range of transaction ids for this body
-    pub fn tx_id_range(&self) -> Range<u64> {
-        self.start_tx_id..self.start_tx_id + self.tx_count
+impl StoredBlockMeta {
+    /// Return the range of transaction ids for this block.
+    pub fn tx_num_range(&self) -> Range<TxNumber> {
+        self.first_tx_num..self.first_tx_num + self.tx_count
+    }
+
+    /// Return the range of transition ids for this block.
+    pub fn transition_range(&self) -> Range<TransitionId> {
+        self.transition_at_block()..self.transition_after_block()
+    }
+
+    /// Return transition id of the state after block executed.
+    pub fn transition_after_block(&self) -> TransitionId {
+        self.first_transition_id + self.tx_count + (self.has_block_change as u64)
+    }
+
+    /// Return transition id of the state at the block execution.
+    pub fn transition_at_block(&self) -> TransitionId {
+        self.first_transition_id
```

Comment:
shouldn't this be the last transition?

Reply:
we now have access to both the first and last transition in one place.
so `transition_at_block` is the **first** transition id and represents "state" before executing the block.
and `transition_after_block` represent the first next transition of the block (first next block transition).

I used this naming as the transition is a strange thing and as it is an index of between the state changes.

Reply:
We should add a clear 1-2 paragraph explanation of how the transition ids work somewhere..

Reply:
I think this is good place, will add good explanation here.
btw about the naming, when it was `BlockBody` i was expecting the transactions when I first saw it, but it contains only indices, and here `BlockMeta` made an entrance. Can I rename it to `BlockBodyIndices` to show it contains only indexes to the other data.

Reply:
Yeah I like that. `BlockBodies` was also confusing to me and I was thinking i'd find transactions inside of it, so I like the idea of calling it Indices.

Reply:
renamed it